service/history/handler.go

```diff
@@ -818,15 +818,12 @@ func (h *Handler) SignalWithStartWorkflowExecution(ctx context.Context, request
 
 		// Two simultaneous SignalWithStart requests might try to start a workflow at the same time.
 		// This can result in one of the requests failing with one of two possible errors:
-		//    1) If it is a brand new WF ID, one of the requests can fail with WorkflowExecutionAlreadyStartedError
-		//       (createMode is persistence.CreateWorkflowModeBrandNew)
-		//    2) If it an already existing WF ID, one of the requests can fail with a CurrentWorkflowConditionFailedError
-		//       (createMode is persisetence.CreateWorkflowModeWorkflowIDReuse)
+		//    CurrentWorkflowConditionFailedError || WorkflowConditionFailedError
 		// If either error occurs, just go ahead and retry. It should succeed on the subsequent attempt.
 		// For simplicity, we keep trying unless the context finishes or we get an error that is not one of the
 		// two mentioned above.
-		_, isExecutionAlreadyStartedErr := err2.(*persistence.WorkflowExecutionAlreadyStartedError)
-		_, isWorkflowConditionFailedErr := err2.(*persistence.CurrentWorkflowConditionFailedError)
+		_, isCurrentWorkflowConditionFailedErr := err2.(*persistence.CurrentWorkflowConditionFailedError)
+		_, isWorkflowConditionFailedErr := err2.(*persistence.WorkflowConditionFailedError)
```

Comment:
These would both be more robust with `errors.As` so that we could safely chain errors.

Reply:
if you plan to use `errors.As` and `errors.Is` then let us talk.

error wrapping in SDK is different then error handling in server.
server persistence layer & dependency wrapper need to convert dependency error to known error type for business logic to handle.